src/functions/sendCandidateResults/application/service/send-notification.ts

```
@@ -0,0 +1,109 @@
+          {address_line_4: notificationPersonalisation.address_line_4} : null),
+        ...(notificationPersonalisation.address_line_5 ?
+          {address_line_5: notificationPersonalisation.address_line_5} : null),
+        ...(notificationPersonalisation.address_line_6 ?
+          {address_line_6: notificationPersonalisation.address_line_6} : null),
+        postcode: notificationPersonalisation.postcode,
+      };
+
+      personalisation = {
+        ...personalisation,
+        ...address,
+      };
+    }
+
+    const renderedText = getRenderedText(testOutcome, notificationPersonalisation, conductedLanguage, padi);
+
+    personalisation = {
+      ...personalisation,
+      renderedText,
+    };
+
+    communicationMethod === Correspondence.EMAIL ?
+      await client.sendEmail(templateId, emailAddress, {personalisation, reference, emailReplyToId}) :
+      await client.sendLetter(templateId, {personalisation, reference});
+    return Promise.resolve();
+  } catch (err: any) {
+    const axiosError = err as unknown as AxiosError;
+    const statusCode = get(axiosError, 'response.status', 0);
+    const message = get(axiosError, 'response.data.errors[0].message', '');
+
+    if (statusCode === 400 || statusCode === 403 || statusCode === 429) {
+      return Promise.reject(new DocumentsServiceError(statusCode, message, false));
+    }
+
+    return Promise.reject(new DocumentsServiceError(statusCode, message, true));
+  }
+}
+
+export function getRenderedText(
+  testOutcome: TestOutcome,
+  notificationPersonalisation: PersonalisationDetails,
+  conductedLanguage: Language,
+  padi: boolean | undefined
+): string | undefined {
+  try {
+    const compileTemplate = Handlebars.compile(
+      templateMapper(testOutcome, notificationPersonalisation.category, conductedLanguage, padi)
+    );
+    return compileTemplate(notificationPersonalisation);
+
+  } catch (error) {
+    console.error('Error preparing renderedText', error);
```

Can we throw this error?